src/base/utils/net.cpp

```diff
@@ -38,5 +40,49 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        bool canParseSubnet(const QString &subnetStr)
+        {
+            Subnet subnet;
+            return canParseSubnet(subnetStr, subnet);
+        }
+
+        bool canParseSubnet(const QString &subnetStr, Subnet subnet)
+        {
+            const Subnet invalid = qMakePair(QHostAddress(), -1);
+            subnet = QHostAddress::parseSubnet(subnetStr);
+            return subnet != invalid;
+        }
+
+        bool isLoopbackAddress(const QHostAddress &addr)
+        {
+            return addr == QHostAddress::LocalHost
+                    || addr == QHostAddress::LocalHostIPv6
+                    || addr == QHostAddress(QLatin1String("::ffff:127.0.0.1"));
+        }
+
+        bool isIPInRange(const QHostAddress &addr, const QList<Subnet> &subnets)
+        {
+            QHostAddress protocolEquivalentAddress;
+            bool addrConversionOk = true;
```

Comment:
please write:
```c++
bool addrConversionOk = false;

if (addr.protocol() == QAbstractSocket::IPv4Protocol) {
    protocolEquivalentAddress = QHostAddress(addr.toIPv6Address());
    addrConversionOk = true;
}
else {
    // only succeeds when addr is an ipv4-mapped ipv6 address
    protocolEquivalentAddress = QHostAddress(addr.toIPv4Address(&addrConversionOk));
}
```

Reply:
There was an explicit reason I had to default this to false but I can't remember it now. Let me mull this one over.